map/src/map/layers/LocalClientTrackLayer.js

```diff
@@ -67,6 +67,20 @@ export default function LocalClientTrackLayer() {
                 checkUpdateLayers();
             }
         }
+        if (ctx.selectedGpxFile?.layers) {
+            /*
+                map.off('click');
+                map.on('click', clickMap);
+
+                clickMap() uses context vars?
+                Got old context inside event handler?
+
+                1. Use Effect() to monitor context changes.
+                2. Disable previous handler (don't queue handlers).
+                3. Re-setup new event handler with refreshed context.
+             */
+            addClickOnMap(); // refresh ctx
```

Comment:
Wrong place to add click. Need add click only in https://github.com/osmandapp/web/blob/d82e763f18863f1238629acc366ca8669af1d50e/map/src/map/layers/LocalClientTrackLayer.js#L106
We add click when start drawing and stop click when finish drawing.

Reply:
resolved by full refactor add/del click (now both are in one place)